libs/ml/include/ml/ops/matrix_multiply.hpp

```diff
@@ -46,22 +42,11 @@ class MatrixMultiply : public fetch::ml::ops::Ops<T>
     , transpose_b_(transpose_b)
   {}
 
-  explicit MatrixMultiply(SPType const &sp)
-    : Ops<T>(sp)
-  {
-    transpose_a_ = sp.transpose_a;
-    transpose_b_ = sp.transpose_b;
-  }
+  explicit MatrixMultiply(SPType const &sp);
 
   ~MatrixMultiply() override = default;
 
-  std::shared_ptr<OpsSaveableParams> GetOpSaveableParams() override
-  {
-    auto ret         = std::make_shared<SPType>();
-    ret->transpose_a = transpose_a_;
-    ret->transpose_b = transpose_b_;
-    return ret;
-  }
+  std::shared_ptr<OpsSaveableParams> GetOpSaveableParams() override;
```

Comment:
it looks like you forgot to move the implementation for MakeSharedCopy into source?

Reply:
Possibly so

Reply:
Done